fr.handbrake.ghb.Plugin.IntelMediaSDK.json

```diff
@@ -0,0 +1,119 @@
+{
+    "app-id": "fr.handbrake.ghb.Plugin.IntelMediaSDK",
+    "branch": "1",
```

Comment:
`  "branch": "stable"`

Reply:
There is very little information to be found on extension development.  But I stumbled upon this at one time https://blog.tingping.se/2018/03/18/flatpaking-plugins.html

It says that `"branch"` in a plugin is a reference back to `"version"` in the `"add-extensions"` section of the apps manifest.  It allows you to manage cases where you may need to maintain multiple versions of a plugin with different APIs or ABIs.

Reply:
If you look at that manifest now (https://github.com/flathub/io.github.Hexchat/blob/master/io.github.Hexchat.json), it uses the version of the freedesktop-sdk that gnome-sdk is based on.

Reply:
The one extension uses `"sdk": "org.freedesktop.Sdk//19.08"`:
https://github.com/flathub/io.github.Hexchat.Plugin.OTR/blob/master/io.github.Hexchat.Plugin.OTR.json

Reply:
I'm afraid that doesn't clear things up for me at all.  

I did some testing to try to understand how different values might affect things.  It definitely does not provide a means of maintaining different plugin versions as the article said since the plugin will still load when the `branch` does not match the `app-extensions` `version`. 

The examples don't really tell me *why* these fields exist and *what* they reference.

Reply:
https://docs.flatpak.org/en/latest/flatpak-command-reference.html?highlight=merge-dirs#flatpak-metadata

> `version` (string)
> 
> The branch to use when looking for the extension. If this is not specified, it defaults to the branch of the application or runtime that the extension point is for. Available since 0.4.1.

Reply:
I somehow did my previous test wrong.  If the `version` in the app does not match the `branch` in the plugin, the plugin does not load.  So the article was correct in it's description of how these fields relate.

So now let me explain why I prefer to use a simple number that is unrelated to SDK versions for these fields.  We do nightly builds of the app and plugin and post bundles for download.  Generally the nightly build of the plugin will be compatible with the release version of the app.  So under usual circumstances, the `branch` in our nightly build plugin should match the `version` in the last release app (as well as the `version` in the nightly app).  But I anticipate that every once in a while there will be a breaking change in development code that will require *bumping* the `version` in the nightly build app and `branch` in the nightly build plugin.

So this is why I went with a simple number for `version` and `branch`.  When a break occurs, I bump both of these in development sources which shows up in the nightly build bundles.  This will ensure that the now incompatible nightly build plugin *does not* load with the release app. At the next release this `version`/`branch` gets propagated to the release and everything is in sync again until the next break in API.